### DISCUSSION TITLE
search result web part in sharepoint

### DISCUSSION THREAD

**Discussion by Ashwinch** [microsoft-search/pnp-modern-search#2322] [Q&A]
I am using search result and search filter webpart, I tried searching for 'cell phone' as an example, I do get the result. When I use handbooks as a option from the search filter under document type, I do get the result, but those handbooks come with unique summary and cannot be opened. Is there a way to make that handbook result open.

The pic shows the handbook results and when I click on it, it renders blank pages. 
The query which I used under query template is: {searchTerms} Path:{siteCollection.URL}

![image](https://user-images.githubusercontent.com/35324820/187397683-7ea09030-40fc-4a65-a3d9-1b5f1e509066.png)

**Comment by Sigi70** [comment]
What content types  are your results? pages?

**Reply by Ashwinch** [reply]
Contents types are handbook chapters or site pages.

**Comment by wobba** [comment]
What does the links look like which you click?

**Reply by Ashwinch** [reply]
As of now, the links are getting directed to correct page.
I changed the result display type to simple list, it displays results along with preview next to it.
however, the handbook does not display any preview and if we click on that preview it shows blank as well.
If we click on preview of word documents, it will display a small pop up where you can scroll down to see the whole document.
Is there a way to do that with sitepages as well?

**Reply by Ashwinch** [reply]
This is how it looks for word or pdf document: 
![image](https://user-images.githubusercontent.com/35324820/188549383-56995a94-46cc-4182-a0b9-8659354d8237.png)

This is how it looks for handbook/sitepage: 
![image](https://user-images.githubusercontent.com/35324820/188549471-065dc942-55c7-4c0b-b948-9a36761399d6.png)

Is there a way to display description like how its showing for those pdf files and also to have that preview work?

**Reply by wobba** [reply]
When I use the list template, and click the thumbnail for a page I do get the actual page showing in the frame. First the thumbnail, then a spinner until the page shows. So not sure what you are experiencing. Was testing with the latest version.

**Reply by Ashwinch** [reply]
Is it working for handbook result as well? Can you share me some images how exactly its displaying for handbook results on your end?

**Reply by wobba** [reply]
There is no concept of "handbook" as a product feature afaik. Is your handbook pages or documents?

**Reply by Ashwinch** [reply]
Those are site pages named as handbook pages.

**Reply by Ashwinch** [reply]
I realized, if we have an image as a banner, it will be displayed.
However, that is not what i want, Is there a way to have a thumbnail like how its showing for the pdf files.
I want to display the content of the page.

**Reply by wobba** [reply]
@Ashwinch there is no API to generate page preview images afaik.
